subprojects/plugins/src/main/java/org/gradle/api/plugins/jvm/internal/DefaultJvmTestSuite.java

```diff
@@ -54,27 +73,63 @@ public DefaultJvmTestSuite(String name, ConfigurationContainer configurations, D
         Configuration implementation = configurations.getByName(sourceSet.getImplementationConfigurationName());
         Configuration runtimeOnly = configurations.getByName(sourceSet.getRuntimeOnlyConfigurationName());
 
-        if (!getName().equals(JvmTestSuitePlugin.DEFAULT_TEST_SUITE_NAME)) {
-            dependencies.addProvider(implementation.getName(), getTestingFramework().map(framework -> {
-                if (framework instanceof JUnitPlatformTestingFramework) {
-                    return "org.junit.jupiter:junit-jupiter-api:" + framework.getVersion().get();
-                } else {
-                    return "junit:junit:" + framework.getVersion().get();
-                }
-            }));
-            dependencies.addProvider(runtimeOnly.getName(), getTestingFramework().map(framework -> {
-                if (framework instanceof JUnitPlatformTestingFramework) {
-                    return "org.junit.jupiter:junit-jupiter-engine:" + framework.getVersion().get();
-                } else {
-                    return getObjectFactory().fileCollection();
-                }
-            }));
+        this.attachedDependencies = false;
+        // This complexity is to keep the built-in test suite from automatically adding dependencies
+        // unless a user explicitly calls one of the useXXX methods
+        // Eventually, we should deprecate this behavior and provide a way for users to opt out
+        // We could then always add these dependencies.
+        this.attachDependencyAction = x -> attachDependenciesForTestFramework(dependencies, implementation);
+
+        if (!name.equals(JvmTestSuitePlugin.DEFAULT_TEST_SUITE_NAME)) {
+            useJUnitJupiter();
+        } else {
+            // for the built-in test suite, we don't express an opinion, so we will assume JUnit4 and not add any dependencies
```

Comment:
```suggestion
            // for the built-in test suite, we don't express an opinion, so we will not add any dependencies
```

Reply:
We don't actually do anything to assume JUnit 4, right?  I was a little confused by this comment

Reply:
yep, that's a better explanation. I was getting ahead of myself